### DISCUSSION TITLE
Goalkeepers Goals Conceded

### DISCUSSION THREAD

**Discussion by ezgal87** [dcaribou/transfermarkt-datasets#190] [Q&A] [answered]
Hej all,
I expcted the GK goal conceded in the appearance dataset, btw I cannot find it.
I'm sure that the datas are available in transfermarkt, and it's quite strange not to found them.
Am I missing something?

ThankS!

**Comment by dcaribou** [comment]
Hi @ezgal87. Thanks for your question.

Not every single attribute that is available in transfermarkt goes to the dataset automatically. There are entities and attributes that are scraped and entities and attributes that are not. Can you provide a screenshot of transfermarkt where this information is available?

In the meantime, I feel you should be able to combine `appearances` and `game_events` to infer conceded goals by the goalkeeper in every appearance.

**Comment by ezgal87** [comment]
Hej @dcaribou , I guess I found the logic behind. 
I'm really sorry but I'm not able to do anything in Python, but in general what I believe is that when it comes to appearances, the scraper collects data from the Compact version, if you switch to the Extended you retrieve also these kind of info.

[Here the example](https://www.transfermarkt.co.uk/sergio-aguero/leistungsdaten/spieler/26399/saison/#gesamt)

From my side it would be really usefull for end-user (like I am, I'm putting your datas in Pbi/Power Automate) that are not really expert with scraping datas.

BR and thank you for the incredible work

**Reply by dcaribou** [reply, marked as answer]
Yes, the link you provided is exactly what gets scraped and added to the dataset. I use the Compact view, not the Extended, just because it happens to be more convenient in some cases but there are very little differences between the two. I don't think any of the two provide the GK conceded goals, just the game aggregate is provided, which you can get from the games file and join to the appearances if your like.

So, basically, you have multiple options
* If you want the aggregate, join to `games` by `game_id`
* If you want GK conceded goals (most of the times this can be inferred from the aggregate, but in some cases this might no be accurate if, for example, the GK is replaced during a game) you can calculate it using `game_events`

You can see the data model with the different files and how to join them in [the README](https://github.com/dcaribou/transfermarkt-datasets/blob/master/README.md).

Also, fyi, the data scraping is done in python, but the data preparation is just SQL. For example, [this](https://github.com/dcaribou/transfermarkt-datasets/blob/master/dbt/models/curated/appearances.sql) is how the appearances file is created.
